paasta_tools/marathon_serviceinit.py

```diff
@@ -127,6 +143,35 @@ def status_marathon_job(service, instance, app_id, normal_instance_count, client
         return "Marathon:   %s - %s (app %s) is %s running in Marathon." % (status, name, app_id, red_not)
 
 
+def marathon_app_deploy_status_human(status, delay=None):
+    status_string = marathon_tools.MarathonDeployStatus.tostring(status)
+
+    if status == marathon_tools.MarathonDeployStatus.Waiting:
+        deploy_status = "%s (new tasks are not launching due to lack of capacity)" % PaastaColors.red(status_string)
+    elif status == marathon_tools.MarathonDeployStatus.Delayed:
+        deploy_status = "%s (next task won't launch for %s seconds due to previous failures)" % (
+                        PaastaColors.red(status_string), delay)
+    elif status == marathon_tools.MarathonDeployStatus.Deploying:
+        deploy_status = PaastaColors.yellow(status_string)
+    elif status == marathon_tools.MarathonDeployStatus.Stopped:
+        deploy_status = PaastaColors.grey(status_string)
+    elif status == marathon_tools.MarathonDeployStatus.Running:
+        deploy_status = PaastaColors.bold(status_string)
+    else:
+        deploy_status = status_string
+
+    return deploy_status
+
+
+def status_marathon_job(service, instance, app_id, normal_instance_count, client):
+    status, delay = marathon_tools.get_marathon_app_deploy_status(app_id, client)
+    deploy_status_human = marathon_app_deploy_status_human(status, delay)
+    if status != marathon_tools.MarathonDeployStatus.NotRunning:
+        running_instances = client.get_app(app_id).tasks_running
```

Comment:
rather than doing this, please always set the ``running_instances`` count, and make it 0 when the status is ``marathon_tools.MarathonDeployStatus.NotRunning``:

```
running_instances = 0 if status == marathon_tools.MarathonDeployStatus.NotRunning else client.get_app(app_id).tasks_running 
```